CLI: revamp team settings to include showcase

mlsteele opened the pull request:
Add support for showcasing to settings. Also rejigger settings command to accommodate more subcommands.

Does this look easy to use?
Do you think that any of these require confirmation prompts? I was thinking they don't.

Examples:
```
$ kbu team settings
Error parsing command line arguments: team name argument required

NAME:
   keybase team settings - Edit team settings.

USAGE:
   keybase team settings [command options] <team name>

OPTIONS:
   -p, --print			Print all your team settings
   --description 		Set the team description
   --open-team 			[reader|writer|off] Set whether anyone can join without being invited and the role they become
   --profile-promote 		[yes|no] Set whether your own profile should promote this team and its description
   --allow-profile-promote 	[yes|no] Set whether non-admins are allowed to promote this team and its description on their profiles
   --showcase 			[yes|no] Set whether to promote this team and its description on keybase.io/popular-teams

$ kbu team settings -p
Error parsing command line arguments: team name argument required

NAME:
   keybase team settings - Edit team settings.

USAGE:
   keybase team settings [command options] <team name>

OPTIONS:
   -p, --print			Print all your team settings
   --description 		Set the team description
   --open-team 			[reader|writer|off] Set whether anyone can join without being invited and the role they become
   --profile-promote 		[yes|no] Set whether your own profile should promote this team and its description
   --allow-profile-promote 	[yes|no] Set whether non-admins are allowed to promote this team and its description on their profiles
   --showcase 			[yes|no] Set whether to promote this team and its description on keybase.io/popular-teams

$ kbu team settings cn3team3 -p
Current settings for team "cn3team3":
  Description:     w
  Open:            no
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        yes (on your profile)

$ kbu team settings cn3team3
Current settings for team "cn3team3":
  Description:     w
  Open:            no
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        yes (on your profile)

$ kbu team settings cn3team3 --open-team false
Team is already closed.
Current settings for team "cn3team3":
  Description:     w
  Open:            no
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        yes (on your profile)

$ kbu team settings cn3team3 --open-team writer
Team set to open.
Current settings for team "cn3team3":
  Description:     w
  Open:            yes
  New member role: writer
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        yes (on your profile)

$ kbu team settings cn3team3 --showcase yes
Current settings for team "cn3team3":
  Description:     w
  Open:            yes
  New member role: writer
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        yes (on your profile)

$ kbu team settings cn3team3 --profile-promote yes
Current settings for team "cn3team3":
  Description:     w
  Open:            yes
  New member role: writer
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        yes (on your profile)

$ kbu team settings cn3team3 --profile-promote no
Current settings for team "cn3team3":
  Description:     w
  Open:            yes
  New member role: writer
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        no

$ kbu team settings cn3team3 --allow-profile-promote yes
▶ ERROR The ability to allow non-admins to promote your team is coming soon!

$ kbu team settings cn3team3 --open-team writer
Team is already open with default role: writer.
Current settings for team "cn3team3":
  Description:     w
  Open:            yes
  New member role: writer
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        no

$ kbu team settings cn3team3 --description "great team"
Current settings for team "cn3team3":
  Description:     great team
  Open:            yes
  New member role: writer
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        no

$ kbu team settings cn3team3 --description ""
Current settings for team "cn3team3":
  Description:
  Open:            yes
  New member role: writer
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        no

$ kbu team settings cn3team3 --description "ok team" --open-team reader
Error parsing command line arguments: only one of these actions a time: description, open-team

NAME:
   keybase team settings - Edit team settings.

USAGE:
   keybase team settings [command options] <team name>

OPTIONS:
   -p, --print			Print all your team settings
   --description 		Set the team description
   --open-team 			[reader|writer|off] Set whether anyone can join without being invited and the role they become
   --profile-promote 		[yes|no] Set whether your own profile should promote this team and its description
   --allow-profile-promote 	[yes|no] Set whether non-admins are allowed to promote this team and its description on their profiles
   --showcase 			[yes|no] Set whether to promote this team and its description on keybase.io/popular-teams

$ kbu team settings cn3team3 --description
Error parsing command line arguments: team name argument required

NAME:
   keybase team settings - Edit team settings.

USAGE:
   keybase team settings [command options] <team name>

OPTIONS:
   -p, --print			Print all your team settings
   --description 		Set the team description
   --open-team 			[reader|writer|off] Set whether anyone can join without being invited and the role they become
   --profile-promote 		[yes|no] Set whether your own profile should promote this team and its description
   --allow-profile-promote 	[yes|no] Set whether non-admins are allowed to promote this team and its description on their profiles
   --showcase 			[yes|no] Set whether to promote this team and its description on keybase.io/popular-teams

$ kbu team settings cn3team3 --open-team
Error parsing command line arguments: team name argument required

NAME:
   keybase team settings - Edit team settings.

USAGE:
   keybase team settings [command options] <team name>

OPTIONS:
   -p, --print			Print all your team settings
   --description 		Set the team description
   --open-team 			[reader|writer|off] Set whether anyone can join without being invited and the role they become
   --profile-promote 		[yes|no] Set whether your own profile should promote this team and its description
   --allow-profile-promote 	[yes|no] Set whether non-admins are allowed to promote this team and its description on their profiles
   --showcase 			[yes|no] Set whether to promote this team and its description on keybase.io/popular-teams
```

zapu commented:
You probably need to rebase to get around prettify bug. Otherwise looks good! Thank you for doing this, cli code is always tedious to write.

malgorithms commented:
it looks good and easy to use. minor preferences:
​
1. in this section of `-p` output:
​
```
  Open:            yes
  New member role: writer
​```

2. Since these are set together as one setting, and so people can understand the latter is for open team joining specifically, you might merge them to:

```​
Open:  yes (default membership=reader)
```

Not a strong preference though.
​
3. In the `-p` output there's no mention of the setting for allow-profile-promote. Is that on purpose or an oversight?
​
4. this error looks wrong (it has a team name, so I can't tell the problem...)
​
```
$ kbu team settings cn3team3 --description
Error parsing command line arguments: team name argument required
​```

5. Is it easy for you to add an example? If so, maybe one that sets two at once:
​
```
EXAMPLE:
   keybase team settings projectx --showcase=yes --profile-promote=yes
actually I'll paste this in the PR discussion
```

In answer to your q, I don't think these require confirmations.

mlsteele commented:
Thanks!

1+2. Changed
```
$ kbu team settings cn3team3 --open-team writer
Team set to open.
Current settings for team "cn3team3":
  Description:
  Open:            yes (default membership = writer)
  Showcased:       yes (on keybase.io/popular-teams)
  Promoted:        yes (on your profile)
```

3. That's on purpose for now because we don't have support yet. Michal's working on that in CORE-6550. The setter flag is also not hooked up. He'll be done so soon I think it's ok to leave it this way.

```
$ kbu team settings cn3team3 --allow-profile-promote yes
▶ ERROR The ability to allow non-admins to promote your team is coming soon!
```

4. yeahbut

The error is wrong and I can't figure a sane way to fix it. We are overdue for a new cli parser.
```
$ kbu team settings cn3team3 --description
Error parsing command line arguments: team name argument required
Yields this data from the arg parsers:
args -> []
kwargs -> {description: "cn3team3"}
```

5. Added some examples

```
$ kbu team settings
Error parsing command line arguments: team name argument required

NAME:
   keybase team settings - Edit team settings.

USAGE:
   keybase team settings [command options] <team name>

DESCRIPTION:
   "keybase team settings" lets you edit settings for a team

EXAMPLES:
Review team settings:
    keybase team settings acme
Open a team so anyone can join as a reader:
    keybase team settings acme --open-team=reader
Showcase a team publicly:
    keybase team settings acme --showcase=yes
Promote a team on your profile:
    keybase team settings acme --profile-promote=yes


OPTIONS:
   -p, --print                  Print all your team settings
   --description                Set the team description
   --open-team                  [reader|writer|off] Set whether anyone can join without being invited and the role they become
   --profile-promote            [yes|no] Set whether your own profile should promote this team and its description
   --allow-profile-promote      [yes|no] Set whether non-admins are allowed to promote this team and its description on their profiles
   --showcase                   [yes|no] Set whether to promote this team and its description on keybase.io/popular-teams
```
